src/drop-down/drop-down.component.ts

```diff
@@ -0,0 +1,375 @@
+                }
+            }
+        }
+
+        return items;
+    }
+
+    /**
+     * Get all header items
+     */
+    public get headers(): IgxDropDownItemComponent[] {
+        const headers: IgxDropDownItemComponent[] = [];
+        if (this.children !== undefined) {
+            for (const child of this.children.toArray()) {
+                if (child.isHeader) {
+                    headers.push(child);
+                }
+            }
+        }
+
+        return headers;
+    }
+
+    /**
+     * Get dropdown html element
+     */
+    public get element() {
+        return this.elementRef.nativeElement;
+    }
+
+    constructor(
+        private elementRef: ElementRef,
+        private cdr: ChangeDetectorRef,
+        private selectionAPI: IgxSelectionAPIService) { }
+
+    /**
+     * Select an item by index
+     * @param index of the item to select
+     */
+    setSelectedItem(index: number) {
+        if (index < 0 || index >= this.items.length) {
+            return;
+        }
+
+        const newSelection = this.items.find((item) => item.index === index);
+        if (newSelection.isDisabled) {
+            return;
+        }
+
+        this.changeSelectedItem(newSelection);
+    }
+
+    /**
+     * Opens the dropdown
+     */
+    open(fireEvents?: boolean) {
+        this.toggleDirective.open(true);
+    }
+
+    /**
+     * Closes the dropdown
+     */
+    close(fireEvents?: boolean) {
+        this.toggleDirective.close(true);
+    }
+
+    /**
+     * Toggles the dropdown
+     */
+    toggle(fireEvents?: boolean) {
+        if (this.toggleDirective.collapsed) {
+            this.open();
+        } else {
+            this.close();
+        }
+    }
+
+    focusFirst() {
+        let focusedItemIndex = 0;
+        while (this.items[focusedItemIndex] && this.items[focusedItemIndex].isDisabled) {
+            focusedItemIndex++;
+        }
+        if (focusedItemIndex < this.items.length - 1) {
+            if (this._focusedItem) {
+                this._focusedItem.isFocused = false;
+            }
+            this._focusedItem = this.items[focusedItemIndex];
+            this._focusedItem.isFocused = true;
+        }
+    }
+
+    focusLast() {
+        let focusedItemIndex = (this.items.length - 1);
+        while (this.items[focusedItemIndex] && this.items[focusedItemIndex].isDisabled) {
+            focusedItemIndex--;
+        }
+        if (focusedItemIndex < this.items.length) {
+            if (this._focusedItem) {
+                this._focusedItem.isFocused = false;
+            }
+            this._focusedItem = this.items[focusedItemIndex];
+            this._focusedItem.isFocused = true;
+        }
+    }
+
+    focusNext() {
+        let focusedItemIndex = -1;
+        if (this._focusedItem) {
+            focusedItemIndex = this._focusedItem.index;
+        }
+        while (this.items[focusedItemIndex + 1] && this.items[focusedItemIndex + 1].isDisabled) {
```

Comment:
Will this logic work if we have given selected item and two more disabled items which are the end of the list.

Reply:
yes, sure. We have manually tested this and we will add test for this.